src/Eloquent/Model.php

```diff
@@ -515,4 +514,58 @@ public function __call($method, $parameters)
 
         return parent::__call($method, $parameters);
     }
+
+    /**
+     * @inheritdoc
+     */
+    protected function addCastAttributesToArray(array $attributes, array $mutatedAttributes): array
+    {
+        foreach ($this->getCasts() as $key => $castType) {
+            if (! Arr::has($attributes, $key) || Arr::has($mutatedAttributes, $key)) {
+                continue;
+            }
+
+            $originalValue = Arr::get($attributes, $key);
+
+            // Here we will cast the attribute. Then, if the cast is a date or datetime cast
+            // then we will serialize the date for the array. This will convert the dates
+            // to strings based on the date format specified for these Eloquent models.
+            $castValue = $this->castAttribute(
+                $key, $originalValue
+            );
+
+            // If the attribute cast was a date or a datetime, we will serialize the date as
+            // a string. This allows the developers to customize how dates are serialized
+            // into an array without affecting how they are persisted into the storage.
+            if ($castValue !== null && in_array($castType, ['date', 'datetime', 'immutable_date', 'immutable_datetime'])) {
+                $castValue = $this->serializeDate($castValue);
+            }
+
+            if ($castValue !== null && ($this->isCustomDateTimeCast($castType) ||
+                    $this->isImmutableCustomDateTimeCast($castType))) {
+                $castValue = $castValue->format(explode(':', $castType, 2)[1]);
+            }
+
+            if ($castValue instanceof DateTimeInterface &&
```

Comment:
https://www.php-fig.org/psr/psr-2/ is deprecated and not fully described all rules with if conditions, i suggest to create task for using psr-12